src/main/scala/firrtl/transforms/InlineNots.scala

```diff
@@ -0,0 +1,68 @@
+package firrtl
+package transforms
+
+import firrtl.ir._
+import firrtl.Mappers._
+import firrtl.PrimOps.Not
+import firrtl.Utils.isNot
+import firrtl.WrappedExpression._
+
+import scala.collection.mutable
+
+object InlineNotsTransform {
+
+  /** Mapping from references to the [[firrtl.ir.Expression Expression]]s that drive them */
+  type Netlist = mutable.HashMap[WrappedExpression, Expression]
+
+  /** Recursively replace [[WRef]]s with new [[Expression]]s
+    *
+    * @param netlist a '''mutable''' HashMap mapping references to [[firrtl.ir.DefNode DefNode]]s to their connected
+    * [[firrtl.ir.Expression Expression]]s. It is '''not''' mutated in this function
+    * @param expr the Expression being transformed
+    * @return Returns expr with Nots inlined
+    */
+  def onExpr(netlist: Netlist)(expr: Expression): Expression = {
+    expr.map(onExpr(netlist)) match {
+      case e @ WRef(name, _,_,_) =>
+        netlist.get(we(e))
+               .filter(isNot)
```

Comment:
I know you also got this from https://github.com/freechipsproject/firrtl/pull/1264, and I've convinced myself that the bug I hit there cannot manifest with this PR, there is the potential for subtle bugs in the future. As I fixed in https://github.com/freechipsproject/firrtl/pull/1264/commits/1ac62e0930d7b0cd5531ea27206b741fcab666ec, it's probably not sufficient to just filter on `isNot`. If we do further inlining, it's possible that this would cause inlining of complex Expressions that just so happen to have a `Not` wrapping them. I'd follow a similar approach as I did in the linked commit to save a future developer some debugging.